frontend/isoko/src/pages/Authentication/SignUp.tsx

```diff
@@ -266,50 +270,36 @@ const SignUp: React.FC = () => {
                      <StyledInput
                         type="email"
                         placeholder="example@gmail.com"
-                        onChange={(e) => setEmail(e.target.value)}
+                        onChange={(e) => {
+                           setEmail(e.target.value);
+                           setErr('');
+                        }}
                      ></StyledInput>{' '}
                      <br />
                      <StyledLabel> Password</StyledLabel>
                      <StyledInput
                         type="password"
                         placeholder="password"
-                        onChange={(e) => setPassword(e.target.value)}
+                        onChange={(e) => {
+                           setPassword(e.target.value);
+                           setErr('');
+                        }}
                      ></StyledInput>
                      <br />
                      <StyledLabel>Confirm Password</StyledLabel>
                      <StyledInput
                         type="password"
                         placeholder="confirm password"
-                        onChange={(e) => setConfirmPassword(e.target.value)}
+                        onChange={(e) => {
+                           setConfirmPassword(e.target.value);
+                           setErr('');
+                        }}
                      ></StyledInput>{' '}
                      <br />
-                     <StyledLabelListBusiness>
-                        Would you like to list a business on ISOKO right now?
-                     </StyledLabelListBusiness>
-                     <CheckboxInputContainer>
-                        <label>
-                           <StyledCheckboxInput
-                              type="checkbox"
-                              checked={listBusinessState === 'YES'}
-                              onChange={() => setListBusinessState('YES')}
-                           />
-                           Yes
-                        </label>
-                        <label>
-                           <StyledCheckboxInput
-                              type="checkbox"
-                              checked={listBusinessState === 'NO'}
-                              onChange={() => setListBusinessState('NO')}
-                           />
-                           No
-                        </label>
-                     </CheckboxInputContainer>
                   </InputContainer>
                   <br />
-                  <WideButton
-                     primary={checkAllFieldsEntered()}
-                     onClick={onSignup}
```

Comment:
I'm a little confused as to why you chose to run the functions in the order you did. For me, logically the clicking on the sign up button should run the onSignup function which should call checkAllFieldsEntered and checkPasswordsMatch.

Reply:
Ya. I definitely see why the order is a little bit weird but there is a reason to it. The first reason is splitting them up to have smaller, more easily testable functions. The other reason is it's easier to display the correct error information this way. 

By chaining the functions like below: Each function is only able to change [err, setErr], once in any given function.
CheckAllFieldsEntered -> checkPasswordMatch -> onSignup 

By following a flow like below: the onSignup function would be able to change the [err, serErr] state in multiple places. Making testing/debugging a little harder. 

onSignup {
checkAllFieldsEnterd()
checkPasswordMatch()
// onsignup stuff 
...
...
...
}

If this is confusing, or non-optimal, we can definitely talk about this further during next standup.

Reply:
makes sense, thanks for explaining!